text/0686-deprecate-old-manager-capabilities-versions.md

```diff
@@ -0,0 +1,163 @@
+  - 3.13
+
+The versions that are still supported are:
+
+- Component Managers
+  - 3.13
+- Modifier Managers
+  - 3.22
+- Helper Managers
+  - 3.23
+
+## How We Teach This
+
+In general, the guides won't need to be updated as there isn't guide material
+for these manager APIs. We should update the API docs for them to remove the
+deprecated capabilities versions. In addition, we should add deprecation guides
+for each of the deprecated versions.
+
+Guides as follows.
+
+### Component Managers
+
+#### 3.4
+
+Any component managers using the `3.4` capabilities should update to the most
+recent component capabilities that are available, currently `3.13`. In `3.13`,
+the only major change is that update hooks are no longer called by default. If
+you need update hooks, use the `updateHook` capability:
+
+```js
+capabilities({
+  updateHook: true,
+});
+```
+
+### Modifier Managers
+
+#### 3.13
+
+Any modifier managers using the `3.13` capabilities should update to the most
+recent modifier capabilities, currently `3.22`. In `3.22`, the major changes
+are:
+
+1. The modifier definition, associated via `setModifierManager` is passed
+   directly to `create`, rather than a factory wrapper class. Previously, you
+   would access the class via the `class` property on the factory wrapper:
+
+   ```js
+   // before
+   class CustomModifierManager {
+     createModifier(Definition, args) {
+       return new Definition.class(args);
+     }
+   }
+   ```
+
+   This can be updated to use the definition directly:
+
+   ```js
+   // after
+   class CustomModifierManager {
+     createModifier(Definition, args) {
+       return new Definition(args);
+     }
+   }
+   ```
+
+2. Args are both lazy and autotracked by default. This means that in order to
+   track an argument value, you must actually use it in your modifier. If you do
+   not, the modifier will not update when the value changes.
+
+   If you still need the modifier to update whenever a value changes, even if it
+   was not used, you can manually access every value in the modifiers
+   `installModifier` and `updateModifier` lifecycle hooks:
+
+   ```js
+   function consumeArgs(args) {
+     for (let key in args.named) {
+       // consume value
+       args.named[key];
+     }
+
+     for (let i = 0; i < args.positional.length; i++) {
+       // consume value
+       args.positional[i];
+     }
+   }
+
+   class CustomModifierManager {
+     installModifier(bucket, element, args) {
+       consumeArgs(args);
+
+       // ...
+     }
+
+     updateModifier(bucket, args) {
+       consumeArgs(args);
+
+       // ...
+     }
+   }
+   ```
+
+   In general this should be avoided, however, and users who are writing
+   modifiers should instead use the value if they want it to be tracked by the
+   modifier.
+
+## Drawbacks
+
+- There will be some minor churn in the ecosystem as managers update, but this
+  is outweighed by the lowered maintenance burden in general for this
+  functionality.
```

Comment:
FWIW, this is a significant issue. Since updating ember-modifer to use 3.22 capabilities is a breaking change **and** ember-cli does not currently support (not well at least) having multiple versions of a given package in the build we have to do a bit of hoop jumping to accomodate this.

Regardless, I think we should do this, but ^ is a very real issue that will need to be addressed before the actual deprecation lands.